app/code/Magento/Inventory/Model/ResourceModel/SourceItem/MultipleSave.php

```diff
@@ -0,0 +1,117 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Inventory\Model\ResourceModel\SourceItem;
+
+use Magento\Framework\App\ResourceConnection;
+use Magento\Inventory\Setup\InstallSchema;
+use Magento\InventoryApi\Api\Data\SourceItemInterface;
+
+/**
+ * Class MultipleSave
+ */
+class MultipleSave
+{
+    /**
+     * @var ResourceConnection
+     */
+    private $connection;
+
+    /**
+     * SourceCarrierLinkManagement constructor
+     *
+     * @param ResourceConnection $connection
+     */
+    public function __construct(
+        ResourceConnection $connection
+    ) {
+        $this->connection = $connection;
+    }
+
+    /**
+     * Multiple save source items
+     *
+     * @param SourceItemInterface[] $sourceItems
+     * @return void
+     */
+    public function multipleSave(array $sourceItems)
+    {
+        if (!empty($sourceItems)) {
```

Comment:
One point of function leave has better readability for me (and it is better for debugging)